slatedb/src/merge_operator.rs

```diff
@@ -179,6 +179,9 @@ pub(crate) struct MergeOperatorIterator<T: KeyValueIterator> {
     /// Whether to merge entries with different expire timestamps.
     merge_different_expire_ts: bool,
     now: i64,
+    /// A barrier sequence number that supports snapshot reads using this iterator. If not None,
+    /// the iterator ill not merge entries with sequence number greater than this value.
```

Comment:
Just to make sure I have the right mental model: the merges will still be _seen_, right? They'll just be treated as individual rows rather than being compacted down into a single one?

Reply:
Yeah that's right. They'll still be seen by snapshot readers that are using a higher seq num.